net/quagga/src/opnsense/scripts/quagga/quagga.rb

```diff
@@ -0,0 +1,242 @@
+          current_if[:area] = $3
+        when /MTU mismatch detection:(.*)/
+          current_if[:mtu_mismatch_detection] = ($1 == 'enabled')
+        when /Router ID ([^,]+?), Network Type ([^,]+?), Cost: (\d+)/
+          current_if[:router_id] = $1
+          current_if[:network_type] = $2
+          current_if[:cost] = $3.to_i
+        when /Transmit Delay is (\d+) sec, State ([^,]+?), Priority (\d+)/
+          current_if[:transmit_delay] = $1.to_i
+          current_if[:state] = $2
+          current_if[:priority] = $3.to_i
+        when "No designated router on this network"
+          current_if[:designated_router] = nil
+        when /Designated Router \(ID\) ([^,]+?), Interface Address (.*)/
+          current_if[:designated_router] = $1
+          current_if[:designated_router_interface_address] = $2
+        when "No backup designated router on this network"
+          current_if[:backup_designated_router] = nil
+        when /Timer intervals configured, Hello (\d+)s, Dead (\d+)s, Wait (\d+)s, Retransmit (\d+)/
+          current_if[:intervals] = {hello: $1.to_i, dead: $2.to_i, wait: $3.to_i, retransmit: $4.to_i}
+        when /Multicast group memberships: (.*)/
+          current_if[:multicast_group_memberships] = $1.split(" ")
+        when /Hello due in ([\d\.]+|inactive)s?/
+          current_if[:hello_due_in] = $1 == 'inactive' ? $1 : $1.to_f
+        when /Neighbor Count is (\d+), Adjacent neighbor count is (\d+)/
+          current_if[:neighbor_count] = $1.to_i
+          current_if[:adjacent_neighbor_count] = $2.to_i
+        else
+          # make sure there is an array to write in
+          current_if[:unparsed] ||= []
+          current_if[:unparsed] << line
+        end
+      end
+    end
+    interfaces
+  end
+  
+  def database
+    lines = @vtysh.execute("show ip ospf database").lines
+    db = {}
+    heading = ''
+    router = ''
+    router_link_states_area = ''
+    mode = :none
+    qta = nil
+    while line = lines.shift
+      next if line == ''
+      if line[0] == ' ' # heading
+        heading = line.strip
+        case heading
+        when /OSPF Router with ID \(([\.\d]+)\)/
+          router = $1
+          db[router] ||= {}
+          mode = :router
+        when /Router Link States \(Area ([\.\d]+)\)/
+          router_link_states_area = $1
+          db[router]['link_state_area'] ||= {}
+          db[router]['link_state_area'][$1] ||= []
+          mode = :link_state
+          qta = nil
+        when 'AS External Link States'
+          mode = :states
+          db[router]['external_states'] ||= []
+          qta = nil
+        else
+          $stderr.puts "unknown heading"
+        end
+      else
+        if qta == nil
+          case mode
+          when :link_state
+            qta = QuaggaTableReader.new(["Link ID", "ADV Router", "Age", "Seq#", "CkSum", "Link count"])
+          when :states
+            qta = QuaggaTableReader.new(["Link ID", "ADV Router", "Age", "Seq#", "CkSum", "Route\n"])
```

Comment:
Note: this is a dirty hack: The class searches the keyword from the left and since `Route` is included in `Router`, it causes issues. However it is good that it is working.